Beams/Logic/vtkSlicerMlcPositionLogic.cxx

```diff
@@ -0,0 +1,1082 @@
+            -1. * isocenterToMLCDistance, isocenterToMLCDistance);
+          leaf2->SetBounds( 0., mlcLeafLength, boundBegin, boundEnd, 
+            -1. * isocenterToMLCDistance, isocenterToMLCDistance);
+        }
+        else // MLCY
+        {
+          leaf1->SetBounds( boundBegin, boundEnd, -1. * mlcLeafLength, 0.0, 
+            -1. * isocenterToMLCDistance, isocenterToMLCDistance);
+          leaf2->SetBounds( boundBegin, boundEnd, 0.0, mlcLeafLength, 
+            -1. * isocenterToMLCDistance, isocenterToMLCDistance);
+        }
+
+        leaf1->Update();
+        leaf2->Update();
+
+        double side1 = 0.0;
+        if (FindLeafAndTargetCollision( beamNode, leaf1->GetOutputPort(), beamInverseTransformFilter->GetOutputPort(),
+          side1, InitialPos1 - 1., 1, typeMLCX))
+        {
+          table->SetValue( leafPair, 1, side1);
+        }
+        else
+        {
+          vtkWarningMacro("CalculateMultiLeafCollimatorPosition: Side 1 collision hasn't been found");
+        }
+
+        double side2 = 0.0;
+        if (FindLeafAndTargetCollision( beamNode, leaf2->GetOutputPort(), beamInverseTransformFilter->GetOutputPort(),
+          side2, InitialPos2 + 1., 2, typeMLCX))
+        {
+          table->SetValue( leafPair, 2, side2);
+        }
+        else
+        {
+          vtkWarningMacro("CalculateMultiLeafCollimatorPosition: Side 2 collision hasn't been found");
+        }
+
+    } // leafPair for cycle ends
+
+  }
+
+  return true;
+}
+
+//---------------------------------------------------------------------------
+bool
+vtkSlicerMlcPositionLogic::FindLeafAndTargetCollision( vtkMRMLRTBeamNode* vtkNotUsed(beamNode), 
+  vtkAlgorithmOutput* leafOutput, vtkAlgorithmOutput* targetOutput,
+  double& sidePos, double initialPosition, int sideType, bool mlcType, 
+  double maxPositionDistance, double positionStep)
+{
+  int contactMode = 1;
+  auto targetMatrix = vtkSmartPointer<vtkMatrix4x4>::New(); // fixed target matrix
+  targetMatrix->Identity();
+  auto leafTransform = vtkSmartPointer<vtkTransform>::New(); // moving leaf transform
+  leafTransform->Identity();
+  auto collide = vtkSmartPointer<vtkCollisionDetectionFilter>::New();
+  collide->SetInputConnection( 0, leafOutput);
+  collide->SetTransform( 0, leafTransform);
+  collide->SetInputConnection( 1, targetOutput);
+  collide->SetMatrix( 1, targetMatrix);
+  collide->SetBoxTolerance(0.0);
+  collide->SetCellTolerance(0.0);
+  collide->SetNumberOfCellsPerNode(2);
+  if (contactMode == 0)
+  {
+    collide->SetCollisionModeToAllContacts();
+  }
+  else if (contactMode == 1)
+  {
+    collide->SetCollisionModeToFirstContact();
+  }
+  else
+  {
+    collide->SetCollisionModeToHalfContacts();
+  }
+  collide->GenerateScalarsOn();
+
+  // Move the leaf
+  int numSteps = fabs(initialPosition) + fabs(maxPositionDistance) / fabs(positionStep);
+  if (sideType == 1)
+  {
+      if (mlcType) // MLCX
```

Comment:
Only two spaces indentation for each block please

Reply:
Fixed.